### PR TITLE
storages: added PeclMemcachedStorage using pecl-memcached extension

### PR DESCRIPTION
**Pull request by hrach**
PHP 7.0 has only memcached extension available.

### CODE REVIEW

**Comment by hrach** [comment]
Well, it seems this repo is disconnected form travis. cc @dg

File: src/Caching/Storages/PeclMemcachedStorage.php [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+			$this->addServer($host, $port);
+		}
+	}
+
+
+	public function addServer($host = 'localhost', $port = 11211, $timeout = 1)
+	{
+		if ($this->memcache->addServer($host, $port, 1) === FALSE) {
+			$error = error_get_last();
+			throw new Nette\InvalidStateException("Memcached::addServer(): $error[message].");
```
Comment (milo) [inline src/Caching/Storages/PeclMemcachedStorage.php]: Isn't it dangerous? It can kill app when unfortunately fail.

Comment (hrach) [inline src/Caching/Storages/PeclMemcachedStorage.php]: Well, 
- app without cache actually can kill the server, not only one request
- this is copy & past & modify from the current MemcachedStorage.

**Comment by hrach** [comment]
tests are green. the question is. should we do something with the naming. (eg. renamed MemcachedStorage to PeclMemcacheStorage (not the missing D) and create deprecated class that inherits it)

**Comment by dg** [comment]
Both are PECL and both are for Memcached, how they differs?

**Comment by hrach** [comment]
Memcached is newer, maintained and updated for PHP7, also has richer API.

---

Ad naming:
You have once mentined that the class name is not a name of the extension but the storage name. But in this case there is not sultion to solve it, so I suggest:

Renamed the current MemcachedStorage to PeclMemcacheStorage (note the missing D) and create deprecated class that inherits it with the old one name.

So this storage will be an exception, where the storage implementation names are based on the extension name

**Comment by JanJakes** [comment]
Memcache extension is not packaged for PHP7 neither by https://www.dotdeb.org/ nor by `ppa:ondrej/php`. Therefore current MemcachedStorage implementation in Nette can be considered incompatible with PHP7.

**Comment by dg** [comment]
Pecl is really not nice, because everything is pecl…

**Comment by hrach** [comment]
We be make a BC break, rename the old one to the MemcacheStorage, and in the new one trhow better exception if memcache exception is loaded and not the memcached.

**Comment by dg** [comment]
But there is incompatibility… http://php.net/manual/en/book.memcache.php#115666

**Comment by hrach** [comment]
I wrote it's a BC break. Nobody claims these storages are interchangeable. Even there is BC break in memcached 5.6 & 7.0 - they have changed the default serializer.

**Comment by hrach** [comment]
For the naming, we can be inspired by Doctrine: https://github.com/doctrine/cache/tree/e0ef9e9833ca024d251e77e8d82a00ca92b0aa71/lib/Doctrine/Common/Cache

**Comment by hrach** [comment]
@dg, please, let's collaborate your real objections and discuss them now. I have no intention to let this open for months or years, in that case I would do an extension instead. But I strongly believe, this should be in the core/package.

**Comment by milo** [comment]
BC break can be solved by prefixing key. What about to hide `Memcache` and `Memcached` into one storage?

**Comment by hrach** [comment]
@milo I don't think it's a good idea. It's pretty common to reuse the connection returned by getConnection.

**Comment by dg** [comment]
To rename current MemcachedStorage is imho not needed, it will be removed in next Nette (for PHP 7), or it can be replaced with this storage. Currently we can use name as NewMemcachedStorage or something similar. It is like mysql & mysqli.

**Comment by hrach** [comment]
If it's temporary, I like PeclMemcached more. But no objections to NewMemcachedStorage. Better something than nothing. So, pick a name, I will rename it and rebase it :)

**Comment by dg** [comment]
Yes, naming things is hard.

I don't like `Pecl` because it says nothing about how they differ. Moreover they both are PECL. 

New extension uses libMemcached, so what about LibMemcached?

**Comment by hrach** [comment]
Thanks. :+1:

**Pull request closed without merging**